aiohue/v2/controllers/events.py

```diff
@@ -116,56 +120,71 @@ async def __event_reader(self) -> NoReturn:
-                    "get", "eventstream/clip/v2", timeout=10, headers=headers
-                ) as resp:
-                    # update status to connected once we reach this point
-                    self._status = EventStreamStatus.CONNECTED
-                    connect_attempts = 1  # reset on succesfull connect
-                    self._logger.debug("Connected to EventStream")
-                    # messages come in one by line, according to EventStream/SSE specs
-                    # we iterate over the incoming lines in the streamreader.
-                    # To prevent a deadlock waiting for a message while the connection is dead
-                    # we have a simple timeout guard in place.
-                    # If no message is received in 30 minutes, a Timeout error will be raised
-                    # and thus the connection re-established.
-                    iterator = resp.content.__aiter__()
-                    while True:
-                        line = await asyncio.wait_for(
-                            iterator.__anext__(), timeout=30 * 60
-                        )
-                        self.__parse_message(line)
+                # use initial connect timeout of 10 seconds
+                async with timeout(10) as timeout_mgr:
+                    # The aiohttp request itself intentionally has a timeout of 0
+                    # because of the longrunning nature of SSE
+                    async with self._bridge.create_request(
+                        "get", "eventstream/clip/v2", timeout=0, headers=headers
+                    ) as resp:
+                        # update status to connected once we reach this point
+                        self._status = EventStreamStatus.CONNECTED
+                        # if this is a reconnect we update the full state
+                        if connect_attempts > 1:
+                            asyncio.create_task(self.__on_reconnect())
+                        connect_attempts = 1
+                        self._logger.debug("Connected to EventStream")
+                        # read over incoming messages line by line
+                        async for line in resp.content:
+                            # process the message
+                            self.__parse_message(line)
+                            # change the timout to CONNECTION_TIMEOUT when a message is received
+                            timeout_mgr.update(loop.time() + CONNECTION_TIMEOUT)
             except (ClientError, asyncio.TimeoutError) as err:
+                # pass expected connection errors because we will auto retry
                 status = getattr(err, "status", None)
                 if status == 404:
                     raise InvalidAPIVersion from err
                 if status == 403:
                     raise Unauthorized from err
-                # pass all other errors because we will auto retry
+                elif status:
```

Comment:
We raise above if the if blocks match. Also looks a bit weird to connect it to only one of the above blocks.
```suggestion
                if status:
```